interfaces/AW-App/src/app/validation-components/validate-program/validate-program.component.ts

```diff
@@ -15,35 +17,202 @@ import { SessionStorageService } from 'src/app/services/session-storage.service'
+      };
+      options.push(questionOption);
+    }
+
+    return options;
+  }
+
+  private mapLabelByLanguageCode(property: any) {
+    let label = property[this.languageCode];
+
+    if (!label) {
+      label = property[this.fallbackLanguageCode];
+    }
+
+    if (!label) {
+      label = property;
+    }
+
+    return label;
+  }
+
+  private getQuestionByCode(questionCode: string): Question {
+    const result = this.questions.find((question: Question) => {
+      return question.code === questionCode;
+    });
+
+    return result;
+  }
+
+  private getAnswerOptionLabelByValue(options: QuestionOption[], answerValue: string) {
+    const option = options.find((item: QuestionOption) => {
+      return item.value === answerValue;
+    });
+
+    return option ? option.label : '';
+  }
+
+  public inputAnswers($event) {
+    const questionCode = $event.target.name;
+    this.answers[questionCode] = new Answer();
+  }
+
+  public changeAnswers($event) {
+    const questionCode = $event.target.name;
+    const answerValue = $event.target.value;
+
+    const question = this.getQuestionByCode(questionCode);
+    const answer: Answer = {
+      code: questionCode,
+      value: answerValue,
+      label: answerValue,
+    };
+
+    // Convert the answerValue to a human-readable label
+    if (question.answerType === AnswerType.Enum) {
+      answer.label = this.getAnswerOptionLabelByValue(question.options, answerValue);
+    }
+
+    this.answers[questionCode] = answer;
+
+    this.checkAllQuestionsAnswered(this.answers);
+
+  }
+
+  public initialAnswers(answersProgram) {
+    for (const answerItem of answersProgram) {
+      const questionCode = answerItem.attribute;
+      const answerValue = answerItem.answer;
+
+      const question = this.getQuestionByCode(questionCode);
+      const answer: Answer = {
+        code: questionCode,
+        value: answerValue,
+        label: answerValue,
+      };
+
+      // Convert the answerValue to a human-readable label
+      if (question.answerType === AnswerType.Enum) {
+        answer.label = this.getAnswerOptionLabelByValue(question.options, answerValue);
+      }
+      this.answers[questionCode] = answer;
+    }
+
+  }
+
+  private checkAllQuestionsAnswered(answers) {
+    for (const key in answers) {
+      if (answers[key].value === '') {
+        this.allQuestionsAnswered = false;
+        return;
+      }
+    }
+    this.allQuestionsAnswered = true;
+  }
+
+  public change() {
+    this.hasAnswered = false;
+    this.changedAnswers = true;
+  }
+
+  public submit() {
+    if (!this.answers.dob) {
+      this.dobFeedback = true;
+      return;
+    }
+    this.hasAnswered = true;
+    this.changedAnswers = false;
+    this.dobFeedback = false;
+  }
+
   public getPrefilledAnswersProgram() {
     this.programsService.getPrefilledAnswers(this.did, this.programId).subscribe(response => {
       this.answersProgram = response;
+      this.initialAnswers(this.answersProgram);
```

Comment:
What type of object is `answersProgram`? Not really a `Program`, right?

We could rename or get rid of the whole public property altogether?

Reply:
adjusted